server/src/middlewares/auth.js

```diff
@@ -13,28 +13,63 @@ const checkJwt = (req, res, next, prisma) => {
     )
   }
 
-  jwt({
-    secret: jwksRsa.expressJwtSecret({
-      cache: true,
-      rateLimit: true,
-      jwksRequestsPerMinute: 1,
-      jwksUri: `https://${conf.auth0Domain}/.well-known/jwks.json`
-    }),
-    credentialsRequired: true,
-    audience: conf.auth0ClientId,
-    issuer: `https://${conf.auth0Domain}/`,
-    algorithms: ['RS256']
-  })(req, res, next)
+  const authType =
+    req.headers.authorization && req.headers.authorization.split(' ')[0]
+
+  req.authType = authType
+
+  if (authType === 'Bearer') {
+    jwt({
+      secret: jwksRsa.expressJwtSecret({
+        cache: true,
+        rateLimit: true,
+        jwksRequestsPerMinute: 1,
+        jwksUri: `https://${conf.auth0Domain}/.well-known/jwks.json`
+      }),
+      credentialsRequired: true,
+      audience: conf.auth0ClientId,
+      issuer: `https://${conf.auth0Domain}/`,
+      algorithms: ['RS256']
+    })(req, res, next)
+  } else if (authType === 'API') {
+    jwt({
+      secret: (req, payload, done) => {
+        if (req.headers['prisma-service'] !== payload['prisma-service']) {
+          done(
+            new jwt.UnauthorizedError(
+              'wrong-prisma-service',
+              'Wrong prisma-service found in JWT Payload'
+            )
+          )
+        }
+        prisma.query
+          .user({ where: { id: payload['user-id'] } }, '{ id key }')
+          .then(user => done(null, user.key))
+      },
+      credentialsRequired: true,
+      algorithms: ['HS256'],
+      getToken: req =>
+        req.headers.authorization && req.headers.authorization.split(' ')[1]
```

Comment:
The code's needs to specify this option because the `Authorization` is not of the format `Bearer <token>`, right?

Reply:
Exactly, we want to differentiate the Auth0 tokens (for classical auth in the frontend) from the API tokens (for the integrations)